java.base/src/main/java/java/io/UnixFileSystem$_native.java

```diff
@@ -261,6 +263,41 @@ public boolean setPermission(File f, int access, boolean enable, boolean owneron
         }
     }
 
+    public String[] list(File f) {
+        final char_ptr pathPtr = mallocPath(f);
```

Comment:
This should be `ptr<c_char>` - I want to get rid of these reified types soon-ish.

Reply:
sure.  I will change that.  We'll have to have a massive cleanup to get rid of `cast()` though as we make the change.   I intentionally used `ptr<DIR>` here, but that then forced me to cast it as an argument in `readdir` and `closedir`.